### PR TITLE
chore: quantitative diff

### PR DESCRIPTION
**Pull request by M4tteoP**
- Quantitative test is not going to fail even if more false positives are detected
- A comment is added to the PR showing the results of the testing
- Quantitative test is executed only if there are changes under `rules/` folder

### CODE REVIEW

**Comment by M4tteoP** [comment]
Example of printed comment in a PR:

📊 Quantitative test results:
 ⚠️ Quantitative testing detected new false positives
📝 Total false positives: `408` -> `10407`
<details>


  <summary>Diff details</summary>

```

3c3
<   "falsePositives": 408,
---
>   "falsePositives": 10407,
10c10
<     "933160": 1,
---
>     "933160": 10000,
15c15
<   "totalTime": 7355236987
---
>   "totalTime": 9614894260

```
</details>

File: .github/workflows/quantitative.yaml [hunk 1]
```diff
@@ -4,14 +4,20 @@ on:
   push:
     branches:
       - main
+    paths:
+      - 'rules/**'
+      - '.github/workflows/quantitative.yaml'
```
Comment (M4tteoP) [inline .github/workflows/quantitative.yaml:9]: The idea is to execute quantitative tests only if there are changes under  `rules/**` or the job itself

File: .github/workflows/quantitative.yaml [hunk 2]
```diff
@@ -45,6 +45,12 @@ jobs:
           gh release download -R coreruleset/go-ftw "v${{ env.GO_FTW_VERSION }}" \
             -p "ftw_${{ env.GO_FTW_VERSION }}_linux_amd64.tar.gz" -O - | tar -xzvf - ftw
 
+      - name: "Restore Cache"
```
Comment (M4tteoP) [inline .github/workflows/quantitative.yaml:47]: go-ftw looks happy
```
DBG Preparing download of corpus file from https://downloads.wortschatz-leipzig.de/corpora/eng_news_2023_10K.tar.gz
DBG filename eng_news_2023_10K-sentences.txt already exists
```

**Review by fzipi — COMMENTED** [review]
LGTM. Let's run it for some days and circle back to see if we want slight changes on the output.

**Review by fzipi — APPROVED** [review]

**Comment by fzipi** [comment]
@M4tteoP Do you want to merge this one?

**Comment by M4tteoP** [comment]
Yes 💯

**Pull request merged**